### ISSUE TITLE
No input reaction

### ISSUE THREAD

**Issue by soblinskas** [csiqueirasilva/PoEController#27]
So the screen opens up without any errors, bet there's no reaction when I try pressing any of the buttons. Seems like they don't get triggered from the applications point of view. Any ideas?

**Comment by csiqueirasilva** [comment]
maybe its something to do with the drivers. if using a non xbox 360 driver,
it might have unmapped inputs for the gamepad. If I had the time currently,
I would create a tool to identify your joystick and map the inputs. This
would be good to debug issues like this.

**Comment by soblinskas** [comment]
It's an xbox wireless controller and probably has native support in win 10. Any advice on how to debug the input? I'm pretty familiar with JS and Node, just not the whole HID and robotjs thing.

**Comment by csiqueirasilva** [comment]
I think you can add some log calls in

https://github.com/csiqueirasilva/PoEController/blob/master/src/game/Controller.js

 to check whats been read you can register your own callback with

HIDController.addListener('data', cb);

and call the logger function on the data received. to check what
poecontroller is actually doing, check Mode.js. When I first created
PoeController, I tried to add different input modes and developed this
module to control them. After the many updates the game had (and I couldnt
keep up with screen pixel reading to swap modes), I changed the code to
keep it in ARPG mode (its a defined value there). If you can fix this,
please do a PR so I can build a release with your code.

**Comment by KAJed82** [comment]
Based on some minor testing the DATA component that is returned from the HID module seems to never change. I am also using an XBOX one wireless controller. I've configured PIDs 0x2E0 and 0x2FF (first one is wireless the second is wired). When I print out the data chunk it shows this:

{"0":0,"1":128,"2":0,"3":128,"4":0,"5":128,"6":0,"7":128,"8":0,"9":128,"10":0,"11":0,"12":0,"13":0}

**Comment by csiqueirasilva** [comment]
That same data comes from both PIDs at all times? Since youre using the
wireless, im guessing the pid should be 0x2E0 and the data should change as
press some buttons on the controller. This same data could come from 0x2FF
if there is an xbox wired controller detected, even if the controller
itself isn't one. What i'm trying to say is: maybe the driver detected it
as an xbox wired controller but can't properly translate inputs into the
data component.

Em dom, 2 de jun de 2019 às 12:05, KAJed82 <notifications@github.com>
escreveu:

**Comment by KAJed82** [comment]
I tested with a single wired controller and a single wireless controller. Both return the same data as I pasted. It never changes between button presses or analog sticks. Initially I tried wireless but since nothing happened I thought to try it wired instead since I know it can be treated differently (even though XINPUT should be in charge of both). I never tested with more than one controller connected and if I look at the PIDs it tested before finding one it does indeed stop at the one that is correct (since the debug out prints all the PIDs that were tried and failed).
